trackma/engine.py

```diff
@@ -865,28 +867,48 @@ def play_episode(self, show, playep=0):
         if not self.mediainfo.get('can_play'):
             raise utils.EngineError('Operation not supported by current site or mediatype.')
 
+        eps = re.split('-', str(playep))
+        if len(eps) > 1:
+            if eps[0] != '':
+                playep = eps[0]
+            else:
+                playep = 0
+            if eps[1] != '':
+                playto = eps[1]
+            else:
+                playto = show['total']
         try:
             playep = int(playep)
+            playto = int(playto)
         except ValueError:
-            raise utils.EngineError('Episode must be numeric.')
+            raise utils.EngineError('Episode[s] must be numeric or with optional range (eg: 1-3, -3, or - to play all unseen episodes)')
 
         if show:
             playing_next = False
             if not playep:
                 playep = show['my_progress'] + 1
                 playing_next = True
 
-            if show['total'] and playep > show['total']:
-                raise utils.EngineError('Episode beyond limits.')
+            if not playto or playto < playep:
+                playto = playep
+
+            if show['total']:
+                if playep > show['total']:
+                    raise utils.EngineError('Episode beyond limits.')
+                if playto > show['total']:
+                    self.msg.info(self.name, "Play to %i is beyond limits of show %s. Defaulting to total episodes of %s" % (playto, show['title'], show['total']))
+                    playto = show['total']
 
             self.msg.info(self.name, "Getting %s %s from library..." % (show['title'], playep))
-            filename = self.get_episode_path(show, playep)
             endep = playep
 
-            if filename:
+            if self.get_episode_path(show, playep):
                 self.msg.info(self.name, 'Found. Starting player...')
                 arg_list = shlex.split(self.config['player'])
-                arg_list.append(filename)
+                for episode in range(playep, playto+1):
+                   ep = self.get_episode_path(show, episode, error_on_fail=False)
+                   if ep:
+                        arg_list.append(ep)
```

Comment:
I think it would be safer to stop appending episodes after the first unfound episode because this suggested behavior is equivalent to silent failure. If episode 4 was missing but 5+ were not, that episode would not be added to the player's playlist and a user wanting hands-off playback would be likely to miss that this episode was never there. If instead we stopped adding episodes alltogether, the user is implicitly prompted to start investigating once he reaches the end of the playlist.

An alternative would be to abort on any error and request an episode range that can be found. For that, the default playto value needs to be the last available episode, not the show's total according to metadata.